test/IntegrationTests/Microsoft.Testing.Platform.Acceptance.IntegrationTests/Helpers/BaseAcceptance.cs

```diff
@@ -10,11 +14,81 @@ namespace Microsoft.Testing.Platform.Acceptance.IntegrationTests;
 /// </summary>
 public abstract class BaseAcceptanceTests : TestBase
 {
+    internal static string RID { get; private set; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "win-x64" : "linux-x64";
+
+    public static string MSTestCurrentVersion { get; private set; }
+
+    static BaseAcceptanceTests()
+    {
+        foreach (var package in Directory.GetFiles(Constants.ArtifactsPackagesNonShipping, "*MSTest*.nupkg", SearchOption.AllDirectories)
+            .Union(Directory.GetFiles(Constants.ArtifactsPackagesShipping, "*MSTest*.nupkg", SearchOption.AllDirectories)).Distinct())
+        {
+            using FileStream fs = File.OpenRead(package);
+            var packageReader = new PackageArchiveReader(fs);
+            var version = packageReader.NuspecReader.GetVersion();
+            if (version.OriginalVersion is null)
+            {
+                throw new InvalidOperationException("Unexpected null nupkg version");
+            }
+
+            if (MSTestCurrentVersion is null)
+            {
+                MSTestCurrentVersion = version.OriginalVersion;
+            }
+            else
+            {
+                if (version.OriginalVersion != MSTestCurrentVersion)
+                {
+                    throw new InvalidOperationException($"Unexpected different package version found in the package folder {version.OriginalVersion} != {MSTestCurrentVersion}");
+                }
+            }
+        }
+
+        if (MSTestCurrentVersion is null)
+        {
+            throw new InvalidOperationException("Unexpected null nupkg version");
+        }
+    }
+
     protected BaseAcceptanceTests(ITestExecutionContext testExecutionContext, AcceptanceFixture acceptanceFixture)
         : base(testExecutionContext)
     {
         AcceptanceFixture = acceptanceFixture;
     }
 
     public AcceptanceFixture AcceptanceFixture { get; }
+
+    internal static IEnumerable<BuildConfiguration> GetBuildConfiguration()
+    {
+        string[] compilationModes = new[] { "Debug", "Release" };
+        foreach (string compilationMode in compilationModes)
+        {
+            yield return compilationMode == "Debug" ? BuildConfiguration.Debug : BuildConfiguration.Release;
+        }
+    }
+
+    internal static TestArgumentsEntry<(string Tfm, BuildConfiguration BuildConfiguration)> FormatGetBuildMatrixTfmBuildConfigurationEntry(TestArgumentsContext ctx)
+    {
+        var entry = ((string, BuildConfiguration))ctx.Arguments;
+        return new TestArgumentsEntry<(string, BuildConfiguration)>(entry, $"{entry.Item1},{entry.Item2}");
+    }
+
+    internal static IEnumerable<(string Tfm, BuildConfiguration BuildConfiguration)> GetBuildMatrixTfmBuildConfiguration()
```

Comment:
Why are you not simpy returning TestArgumentsEntry here instead of having this method + the formatter method?

Reply:
because this can be used for other composition in future where something more needs to be added ike , verb and formatting is usual "testing method specific".

Reply:
If you prefer single method and no composition it's fine to me.

Reply:
done